dart/constraint/ConstraintSolver.h

```diff
@@ -110,11 +110,17 @@ class ConstraintSolver
 
   /// Set collision detector
   void setCollisionDetector(
-    std::unique_ptr<collision::CollisionDetector>&& _collisionDetector);
+    std::unique_ptr<collision::CollisionDetector> _collisionDetector);
 
   /// Get collision detector
   collision::CollisionDetector* getCollisionDetector() const;
 
+  /// Set LCP solver
+  void setLCPSolver(std::unique_ptr<LCPSolver> _lcpSolver);
```

Comment:
That shouldn't be done for ``unique_ptr`` because we want the ownership of the solver to be passed into the ``ConstraintSolver``. Passing by reference wouldn't allow that.

``unique_ptr`` is really weird when it comes to function arguments.